### PR TITLE
Make sure that logged URL is the same as the one which is requested.

### PR DESCRIPTION
**Pull request by drupol**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #31068 
| License       | MIT

### CODE REVIEW

File: src/Symfony/Component/HttpClient/NativeHttpClient.php [hunk 1]
```diff
@@ -159,6 +159,9 @@ public function request(string $method, string $url, array $options = []): Respo
             $this->multi->dnsCache = $options['resolve'] + $this->multi->dnsCache;
         }
 
+        // We keep a reference of the real URL for the logger.
+        $stringUrl = implode('', $url);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpClient/NativeHttpClient.php]: $realUrl?

File: src/Symfony/Component/HttpClient/NativeHttpClient.php [hunk 2]
```diff
@@ -208,8 +211,7 @@ public function request(string $method, string $url, array $options = []): Respo
         $context = stream_context_create($context, ['notification' => $notification]);
         self::configureHeadersAndProxy($context, $host, $options['request_headers'], $proxy, $noProxy);
 
-        $url = implode('', $url);
-        $this->logger && $this->logger->info(sprintf('Request: %s %s', $method, $url));
+        $this->logger && $this->logger->info(sprintf('Request: %s %s', $method, $stringUrl));
 
         return new NativeResponse($this->multi, $context, $url, $options, $gzipEnabled, $info, $resolveRedirect, $onProgress, $this->logger);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/HttpClient/NativeHttpClient.php]: `s/$url/implode('', $url)` on this line then

**Review by nicolas-grekas — APPROVED** [review]
good catch! just a small comment and good to me

File: src/Symfony/Component/HttpClient/NativeHttpClient.php (hunk 1 above)
Comment (nicolas-grekas) [reply src/Symfony/Component/HttpClient/NativeHttpClient.php]: alternatively, what about logging here?

Comment (drupol) [reply src/Symfony/Component/HttpClient/NativeHttpClient.php]: I though about it as well but I didn't want to make this change yet until I had some feedback first. If you think it's also valid to log up there, it's fine.

File: src/Symfony/Component/HttpClient/NativeHttpClient.php (hunk 2 above)
Comment (drupol) [reply src/Symfony/Component/HttpClient/NativeHttpClient.php]: yes, that was one of my idea if we would have moved up this line.

File: src/Symfony/Component/HttpClient/NativeHttpClient.php (hunk 1 above)
Comment (drupol) [reply src/Symfony/Component/HttpClient/NativeHttpClient.php]: Fixed.

File: src/Symfony/Component/HttpClient/NativeHttpClient.php (hunk 2 above)
Comment (drupol) [reply src/Symfony/Component/HttpClient/NativeHttpClient.php]: Fixed.

**Comment by nicolas-grekas** [comment]
Thank you @drupol.

**Pull request merged**